docs/HONEYBADGER_KO.md

```diff
@@ -0,0 +1,193 @@
+# HoneyBadger BFT Protocol
+### Introduction
+기존의 분산 장애 허용 프로토콜(Distributed fault tolerance protocol)은 주로 작은 규모, 단일 관리 도메인(Single administrative domain)을 
+목표로 배포되고 있었습니다.<br>
+그러나 최근에는 새로운 분산 시스템인 *암호화폐(cryptocurrencies)*, *블록체인(blockchains)* 이 등장하면서, 예측이 힘든 큰 규모의 네트워크에 적합하며, 
+악의적인 노드들에 저항성을 갖는 시스템들에 대한 연구들이 진행되었습니다.<br>
+이런 프로토콜은 아래와 같은 점들을 만족해야 합니다.<br>
+* **견고함**<br>
+암호화폐는 성능을 희생해서라도 견고한 보안을 유지하려고 합니다.<br>
+비트코인(Bitcoin)은 분산 시스템의 기준에서 굉장히 나쁜 성능을 갖고 있지만, 기존의 분산 시스템에 비해 훨씬 더 많은 위험에 노출되어 있기 때문에 견고한 시스템인 
+작업 증명(PoW, Proof of Work)을 선택했습니다.<br>
+
+* **지연 대비 처리량 향상**<br>
+기존의 장애 허용 시스템이 성능 최적화에 많은 관심을 보였다면, 블록체인은 원활한 환불 및 롤백 처리를 위해 의도적으로 트랜젝션 처리에 지연을 두기도 합니다.<br>
+
+### HoneyBadgerBFT의 특징
+#### 비동기적인 원자적 브로드캐스트 프로토콜
+대부분의 Byzantine fault tolerant(BFT) 시스템에서는 메세지가 *정해진 시간 안에 도착* 해야 합니다.<br>
+그러나 HBBFT는 이러한 timeout 기반의 로직은 신뢰하기 힘든 네트워크에서 동작하는 분산 시스템, 또는 암호화폐에 적합하지 않다고 판단합니다.<br>
+따라서 HBBFT는 *비동기적인 원자적 브로드캐스트 프로토콜(asynchronous atomic broadcast protocol)* 을 사용합니다.<br>
+
+#### 블록 생성의 난이도
+HBBFT는 네트워크 자원과 무관하게 계산량이 많은 암호화폐 등에 적합하도록 설계되었습니다.<br>
+이런 특성은 기존의 장애 허용 시스템에서는 어려웠던 블록 생성 작업을 쉽게 해줍니다.<br>
+
+### HoneyBadgerBFT 프로토콜
+HoneyBadger 프로토콜은 네트워크에 N개의 노드(`P(0)`부터 `P(N-1)`까지)를 설정합니다.<br>
+각 노드들은 트랜잭션을 입력받으며, 프로토콜의 목적은 **모든 노드들의 공통의 합의(common consensus)를 이끌어내는 것** 입니다.<br>
+
+원자적 브로드캐스트는 응용 프로그램의 세부 정보(가령, 트랜잭션이 어떻게 해석되는지)를 추상화 할 수 있습니다.<br>
+실제로, 클라이언트는 유일한 문자열로 이루어진 트랜잭션을 생성하여 모든 노드에 전송하고, 대다수의 노드로부터 서명을 수집한 후 커밋했다고 판단합니다.<br>
+
+HoneyBadger 프로토콜은 아래와 같은 특징이 있습니다.<br>
+* 순수 비동기적인 네트워크<br>
+HoneyBadgerBFT의 노드들은 각자 신뢰성있는 p2p 연결로 이루어져 있으며, 메세지를 드롭하지 않습니다.<br>
+
+* 정적인 Byzantine faults<br>
+매개변수로 받은 f만큼 비잔틴 허용 가능하게 합니다.<br>
+이 때, f는 `3f + 1 <= N` 을 만족해야 합니다.
+
+* 신뢰 설정<br>
+각 단계의 공개키 설정 및 비밀 공유를 위한 초기 구성 단계에서는 신뢰할 수 있는 노드와 상호작용 합니다.<br>
+만약 실제 배포에서 신뢰할 수 있는 그룹이 없는 경우에는 분산 키 생성 프로토콜을 대신 이용할 수 있습니다.<br>
+
+#### 원자적 브로드캐스트 (Atomic Broadcast)
+HBBFT는 원자적 브로드캐스트를 다음과 같이 정의합니다.<br>
+* 합의 : 어떤 노드가 트랜잭션 tx를 출력하면, 다른 모든 노드들도 tx를 출력해야 합니다.*<br>
+
+* 전체 순서 : 만약 어떤 노드가 {tx(0), tx(1), ..., tx(i)}의 순서로 트랜잭션을 출력했다면, 다른 모든 노드들도 똑같은 순서로 출력해야 합니다.*<br>
+
+* 검열 저항 : 트랜잭션 tx가 올바른 N-f개의 노드에 입력되면, 결국 모든 올바른 노드에 의해 출력됩니다.*<br>
+
+#### HoneyBadgerBFT의 동작
+노드는 트랜잭션을 입력으로 받고 버퍼에 저장합니다.<br>
+프로토콜은 단계별로 진행되며, 각 단계 이후에는 커밋된 로그에 새로운 트랜잭션 배치가 추가됩니다.<br>
+각 단계의 시작에서 노드들은 각자의 정책에 따라 버퍼에 저장된 트랜잭션의 부분집합을 선택합니다. 그리고 그 부분집합을 무작위 계약 프로토콜 인스턴스에 입력합니다.<br>
+합의 프로토콜의 마지막 단계에서는, 트랜잭션의 최종 집합이 선택됩니다.<br>
+
+### 비동기적인 서브셋에서의 HBBFT 구축
+**비동기적인 서브셋 : ACS(Asynchronous common subset)**<br>
+
+ACS 프로토콜은 다음과 같은 특징을 갖습니다.<br>
+
+* 유효성 : 정상적인 노드가 집합 v를 출력하면, `|v| >= N - f` 를 만족하며 v는 적어도 `N - 2f` 개의 정상적인 노드의 입력를 포함합니다.
```

Comment:
It is better to write it in English because the Korean is awkward. like a `Validity`, `Agreement`,`Totality`.